src/Data/Algorand/Transaction.hs

```diff
@@ -118,29 +113,60 @@ data TransactionType
   deriving (Eq, Generic, Show)
 
 -- | Constants for @OnComplete@.
---
-type OnComplete = Text
-
-onCompleteNoOp :: OnComplete
-onCompleteNoOp = "call"
-
-onCompleteCreate :: OnComplete
-onCompleteCreate = "create"
-
-onCompleteOptIn :: OnComplete
-onCompleteOptIn = "optin"
-
-onCompleteCloseOut :: OnComplete
-onCompleteCloseOut = "closeout"
-
-onCompleteClearState :: OnComplete
-onCompleteClearState = "clear"
-
-onCompleteUpdateApplication :: OnComplete
-onCompleteUpdateApplication = "update"
-
-onCompleteDeleteApplication :: OnComplete
-onCompleteDeleteApplication = "delete"
+data OnComplete
+  = OnCompleteNoOp
+  | OnCompleteOptIn
+  | OnCompleteCloseOut
+  | OnCompleteClearState
+  | OnCompleteUpdateApplication
+  | OnCompleteDeleteApplication
+  deriving (Eq, Show)
+
+instance Default OnComplete where
+  def = OnCompleteNoOp
+instance CanonicalZero OnComplete
+instance NonZeroValue OnComplete
+
+instance Enum OnComplete where
+  fromEnum OnCompleteNoOp = 0
+  fromEnum OnCompleteOptIn = 1
+  fromEnum OnCompleteCloseOut = 2
+  fromEnum OnCompleteClearState = 3
+  fromEnum OnCompleteUpdateApplication = 4
+  fromEnum OnCompleteDeleteApplication = 5
+
+  toEnum 0 = OnCompleteNoOp
+  toEnum 1 = OnCompleteOptIn
+  toEnum 2 = OnCompleteCloseOut
+  toEnum 3 = OnCompleteClearState
+  toEnum 4 = OnCompleteUpdateApplication
+  toEnum 5 = OnCompleteDeleteApplication
+  toEnum x = error ("Expect `OnComplete` integer value to be from 0 - 5 but got: " <> show x)
+
+instance AlgoMessagePack OnComplete where
+  toAlgoObject = toAlgoObject @Word64 . fromIntegral . fromEnum
```

Comment:
We use only 5 possible options for `OnComplete`. So, why do not to use `Word8` for example?

Reply:
Make sense, fixed this.